File: src/main/java/io/openfuture/openmessanger/security/CognitoAuthenticationToken.java
package io.openfuture.openmessanger.security;

import java.util.Collection;

import org.springframework.security.authentication.AbstractAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

public class CognitoAuthenticationToken extends AbstractAuthenticationToken {

    private final String username;
    private final String token;

    public CognitoAuthenticationToken(final String token) {
        super(null);
        this.token = token;
        this.username = null;
    }

    public CognitoAuthenticationToken(final String username,
                                      final String token,
                                      final Collection<? extends GrantedAuthority> authorities) {
        super(authorities);
        super.setAuthenticated(true);
        super.eraseCredentials();
        this.token = token;
        this.username = username;
    }

    @Override
    public Object getCredentials() {
        return this.token;
    }

    @Override
    public Object getPrincipal() {
        return this.username;
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/security/CognitoAuthenticationToken.kt
package io.openfuture.openmessanger.security

import org.springframework.security.authentication.AbstractAuthenticationToken
import org.springframework.security.core.GrantedAuthority

class CognitoAuthenticationToken : AbstractAuthenticationToken {
    private val username: String?
    private val token: String

    constructor(token: String) : super(null) {
        this.token = token
        username = null
    }

    constructor(
        username: String,
        token: String,
        authorities: Collection<GrantedAuthority?>?
    ) : super(authorities) {
        super.setAuthenticated(true)
        super.eraseCredentials()
        this.token = token
        this.username = username
    }

    override fun getCredentials(): Any {
        return token
    }

    override fun getPrincipal(): Any? {
        return username
    }
}